src/rgw/rgw_iam_policy.cc

```diff
@@ -792,6 +792,8 @@ static optional<Principal> parse_principal(CephContext* cct, TokenID t,
 bool ParseState::do_string(CephContext* cct, const char* s, size_t l) {
   auto k = pp->tokens.lookup(s, l);
   Policy& p = pp->policy;
+  bool isAction = false;
```

Comment:
Is there a real need to take isAction variable?
Because whenever it goes inside case `((w->id == TokenID::Action) || (w->id == TokenID::NotAction))` it will loop through the `for()` .
And if it does not go inside `((w->id == TokenID::Action) || (w->id == TokenID::NotAction))` `isValidAction` is always false

Reply:
If we do not mark isAction flag,  will return false when w->id is another value. For example, w->id == TokenID::Effect, isVaildAction is false,. At finally, "if(!isValidAction)return false" will be executed.

Reply:
Sure, looks reasonable,

Reply:
nit, might be consistent with the naming convention: we don't use camelCase in Ceph, so i'd suggest rename this variable to `is_action` instead.